spyder/workers/updates.py

```diff
@@ -140,14 +140,13 @@ def start(self):
 
             result = self.check_update_available()
             self.update_available, self.latest_release = result
-        except URLError:
-            error_msg = _(
-                'It was not possible to connect to the internet to check for '
-                'Spyder updates.'
-                '.<br><br>'
-                'Make sure the connection is working properly.'
-            )
-        except Exception:
+        except SSLError as err:
+            error_msg = SSL_ERROR_MSG
+            logger.debug(err, stack_info=True)
+        except ConnectionError as err:
+            error_msg = CONNECT_ERROR_MSG
+            logger.debug(err, stack_info=True)
+        except Exception as err:
```

Comment:
We could add one more block here to catch `HTTPError`:

```python
except HTTPError as err:
    error_msg = HTTP_ERROR_MSG.format(page.status_code)
    ...
```

with a friendly message, e.g. :

```python
HTTP_ERROR_MSG = _(
    'HTTP error {status_code} when checking for updates.'
    '<br><br>Make sure your connection is working properly.'
)
```